Core/ControlMapper.cpp

```diff
@@ -375,3 +386,55 @@ void ControlMapper::processAxis(const AxisInput &axis, int direction) {
 		}
 	}
 }
+
+void ControlMapper::ProcessAnalogSpeed(const AxisInput &axis, bool opposite) {
+	FPSLimit &limitMode = PSP_CoreParameter().fpsLimit;
+	// If we're using an alternate speed already, let that win.
+	if (limitMode != FPSLimit::NORMAL && limitMode != FPSLimit::ANALOG)
+		return;
+	// Don't even try if the limit is invalid.
+	if (g_Config.iAnalogFpsLimit <= 0)
+		return;
+
+	AnalogFpsMode mode = (AnalogFpsMode)g_Config.iAnalogFpsMode;
+	float value = axis.value;
+	if (mode == AnalogFpsMode::AUTO) {
+		// TODO: Consider the pad name for better auto?  KeyMap::PadName(axis.deviceId);
+		switch (axis.axisId) {
+		case JOYSTICK_AXIS_X:
+		case JOYSTICK_AXIS_Y:
+		case JOYSTICK_AXIS_Z:
+		case JOYSTICK_AXIS_RX:
+		case JOYSTICK_AXIS_RY:
+		case JOYSTICK_AXIS_RZ:
+			// These, at least on directinput, can be used for triggers that go from mapped to opposite.
+			mode = AnalogFpsMode::MAPPED_TO_OPPOSITE;
+			break;
+
+		default:
+			// Other axises probably don't go from negative to positive.
+			mode = AnalogFpsMode::MAPPED_DIRECTION;
+			break;
+		}
+	}
+
+	// Okay, now let's map it as appropriate.
+	if (mode == AnalogFpsMode::MAPPED_DIRECTION) {
+		value = fabsf(value);
+		if (opposite)
+			return;
```

Comment:
This does indeed help:

```c
		value = fabsf(value);
		if (opposite) {
			value = 0.0f;
		}
```
instead of returning.

Though, I would argue that the code that tries to figure out an "opposite" key for a throttle already is wrong, but that's a separate bug. The value that comes in in that case is indeed 0.0, and returning there means that we miss the return to 0.0.

Reply:
Yeah basically if it's a zero value, the mapping finds itself as an opposite, and skips it :)

Reply:
Hm, I guess that's fine, I suppose any case where you don't want the negative (like mapping right analog X to speed) it'd make sense for it to clamp.

So `direction` is -1 for 0?  I was expecting both to get called, because of this:

```
	if (axis.value > 0) {
		processAxis(axis, 1);
		return true;
	} else if (axis.value < 0) {
		processAxis(axis, -1);
		return true;
	} else if (axis.value == 0) {
		// Both directions! Prevents sticking for digital input devices that are axises (like HAT)
		processAxis(axis, 1);
		processAxis(axis, -1);
		return true;
	}
```

So sure, one would get skipped, but the other wouldn't.  I guess it's probably not exactly equaling 0 or something?

-[Unknown]

Reply:
> Code looks good, the MAPPED_TO_OPPOSITE stuff is a bit unintuitively named I feel, but okay I guess.

Well, changed to MAPPED_DIR_TO_OPPOSITE_DIR.  I guess it could be THROUGH or UNTIL or something, but already kinda long.  I guess it could be MAPPED_FULL_RANGE or MAPPED_EDGE_TO_EDGE or something, none of them are great.

-[Unknown]

Reply:
Both do get called, so hm, I retract my statement about a bug, and now I'm slightly confused how this happens. The change does help though! I'm gonna debug some more because this doesn't quite make sense...